tools/kubectl-karpenter/pkg/kubectl-karpenter/kubectl-karpenter.go

```diff
@@ -0,0 +1,161 @@
+	Printer    printers.ResourcePrinter
+
+	builder func() *resource.Builder
+	local   bool
+
+	resource.FilenameOptions
+	genericiooptions.IOStreams
+}
+
+func NewCmd(f cmdutil.Factory, ioStreams genericiooptions.IOStreams) *cobra.Command {
+	o := Options{
+		PrintFlags: genericclioptions.NewPrintFlags("converted").WithDefaultOutput("yaml"),
+		local:      true,
+		IOStreams:  ioStreams,
+	}
+
+	var rootCmd = &cobra.Command{
+		Use: "kubectl karpenter",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+		},
+	}
+
+	var convertCmd = &cobra.Command{
+		Use:   "convert",
+		Short: "Convert a file or directory",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd))
+			cmdutil.CheckErr(o.RunConvert())
+		},
+	}
+
+	cmdutil.AddFilenameOptionFlags(convertCmd, &o.FilenameOptions, "to need to get converted.")
+	convertCmd.MarkFlagRequired("file")
+
+	rootCmd.AddCommand(convertCmd)
+
+	return rootCmd
+}
+
+func (o *Options) Complete(f cmdutil.Factory, cmd *cobra.Command) (err error) {
+	err = o.FilenameOptions.RequireFilenameOrKustomize()
+	if err != nil {
+		return err
+	}
+	o.builder = f.NewBuilder
+	o.Printer, err = o.PrintFlags.ToPrinter()
+	return err
+}
+
+func (o *Options) RunConvert() error {
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
+	v1alpha5.SchemeBuilder.AddToScheme(scheme)
+
+	b := o.builder().
+		WithScheme(scheme, v1alpha1.SchemeGroupVersion, v1alpha5.SchemeGroupVersion).
+		LocalParam(o.local)
+
+	r := b.
+		ContinueOnError().
+		FilenameParam(false, &o.FilenameOptions).
+		Flatten().
+		Do()
+
+	err := r.Err()
+	if err != nil {
+		return err
+	}
+
+	singleItemImplied := false
+	infos, err := r.IntoSingleItemImplied(&singleItemImplied).Infos()
+	if err != nil {
+		return err
+	}
+
+	if len(infos) == 0 {
+		return fmt.Errorf("no objects passed to convert")
+	}
+
+	for _, info := range infos {
+		if info.Object == nil {
+			continue
+		}
+
+		obj, err := Process(info.Object)
+		if err != nil {
+			fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+		} else {
+			if err := o.Printer.PrintObj(obj, o.Out); err != nil {
+				fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+func Process(resource runtime.Object) (runtime.Object, error) {
+	kind := resource.GetObjectKind().GroupVersionKind().Kind
+	switch kind {
+	case "Provisioner":
+		return processProvisioner(resource), nil
+	case "AWSNodeTemplate":
+		return processNodeTemplate(resource), nil
+	default:
+		return nil, fmt.Errorf("unknown kind. expected one of Provisioner, AWSNodeTemplate. got %s", kind)
+	}
+}
+
+func processNodeTemplate(resource runtime.Object) runtime.Object {
+	nodetemplate := resource.(*v1alpha1.AWSNodeTemplate)
+
+	nodeclass := nodeclassutil.New(nodetemplate)
+	nodeclass.TypeMeta = metav1.TypeMeta{
+		Kind:       "EC2NodeClass",
+		APIVersion: v1beta1.SchemeGroupVersion.String(),
+	}
+	role := "<your AWS role here>"
+	nodeclass.Spec.Role = &role
```

Comment:
```suggestion
	nodeclass.Spec.Role = lo.ToPtr("<your AWS role here>")
```

nit: You could consider using https://github.com/samber/lo to one-line this